src/tests/suite/daos_obj.c

```diff
@@ -3365,6 +3365,161 @@ io_pool_map_refresh_trigger(void **state)
 	ioreq_fini(&req);
 }
 
+/**
+ * Test to fetch the non existance keys.
+ */
+static void nonexistent_internal(void **state, daos_obj_id_t oid,
+				 const char key_type[])
+{
+	test_arg_t	*arg = *state;
+	struct ioreq	req;
+	char		*update_buf;
+	char		*fetch_buf;
+	unsigned int	size   = IO_SIZE_NVME;
+	const char	dkey[] = "nonexistent dkey";
+
+	D_ALLOC(update_buf, size);
+	assert_non_null(update_buf);
+	dts_buf_render(update_buf, size);
+	ioreq_init(&req, arg->coh, oid, DAOS_IOD_ARRAY, arg);
+	D_ALLOC(fetch_buf, size);
+	assert_non_null(fetch_buf);
+	memset(fetch_buf, 0, size);
+
+	/** create dkey to fetch nonexist akey*/
+	if (strcmp(key_type, "akey") == 0) {
+		insert_single(dkey, "first_akey", 0, update_buf, size,
+			      DAOS_TX_NONE, &req);
+	}
+
+	lookup_single(dkey, "nonexistent akey", 0, fetch_buf, size,
+		      DAOS_TX_NONE, &req);
+
+	/**
+ 	* As per current implementation non existing keys
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs